### PR TITLE
Feat: Split Remote into IP & Port

### PR DESCRIPTION
**Pull request by Blokje5**
Ensure that IP & Port are split in logging for easier analysis.

Signed-off-by: Lennard Eijsackers <lennardeijsackers92@gmail.com>

### CODE REVIEW

**Review by martinvisser — APPROVED** [review]
LGTM

**Comment by blagerweij** [comment]
@Blokje5 What happens with IPv6 addresses, since these contain ':' characters ?

**Comment by blagerweij** [comment]
```
    ip, port, err := net.SplitHostPort(req.RemoteAddr)
    if err != nil {
        //return nil, fmt.Errorf("userip: %q is not IP:port", req.RemoteAddr)

        fmt.Fprintf(w, "userip: %q is not IP:port", req.RemoteAddr)
    }

    userIP := net.ParseIP(ip)
    if userIP == nil {
        //return nil, fmt.Errorf("userip: %q is not IP:port", req.RemoteAddr)
        fmt.Fprintf(w, "userip: %q is not IP:port", req.RemoteAddr)
        return
    }

    // This will only be defined when site is accessed via non-anonymous proxy
    // and takes precedence over RemoteAddr
    // Header.Get is case-insensitive
    forward := req.Header.Get("X-Forwarded-For")
```
If `forward` is defined, we should use that, otherwise use `userIP`

File: jwt.go [hunk 1]
```diff
@@ -384,6 +395,44 @@ func (jwtPlugin *JwtPlugin) ExtractToken(request *http.Request) (*JWT, error) {
 	return &jwtToken, nil
 }
 
+func (jwtPlugin *JwtPlugin) remoteAddr(req *http.Request) Network {
+	// This will only be defined when site is accessed via non-anonymous proxy
+	// and takes precedence over RemoteAddr
+	// Header.Get is case-insensitive
+	ipHeader := req.Header.Get("X-Forwarded-For")
```
Comment (Blokje5) [inline jwt.go:402]: Does X-Forwarded-For actually contain the port as well? I guess it doesn't matter as the port is not the most relevant thing.

File: jwt.go [hunk 2]
```diff
@@ -384,6 +395,44 @@ func (jwtPlugin *JwtPlugin) ExtractToken(request *http.Request) (*JWT, error) {
...
+	if len(ipHeader) == 0 {
+		ipHeader = req.RemoteAddr
+	}
+
+	ip, port, err := net.SplitHostPort(ipHeader)
+	portNumber, _ := strconv.Atoi(port)
```
Comment (Blokje5) [inline jwt.go:408]: We could also make the port a string, given that it is only used for logging anyway.

Comment (martinvisser) [reply jwt.go:408]: Does DataDog interpret them differently?

**Review by blagerweij — APPROVED** [review]

**Pull request merged**